### DISCUSSION TITLE
New Plugin Review & Improvement - Products Carousel

### DISCUSSION THREAD

**Discussion by perks98** [elementor/elementor#22895] [Developers Community]
Hi all,
I have created my first Elementor addon which extends the current Products widget by using the carousel trait allowing the swiper library enabling products to swipe in slides rather than in a grid format. For my background, I don't know 100% what I am doing as this is my first plugin 😅. The use case for this is for websites where you rely on the look and consistent style of the theme made product loop item listing which works better for third-party plugins but has no options for a carousel as Elementor only offers in a grid style for the base Products widget.

What I need help with is for anyone who knows about Elementor addons development to review my plugin to make sure it follows best practices, any suggestions to improve the code and makes sure future Elementor updates do not break it. I would appreciate any help. I have uploaded a zip of the plugin below for you to download and use.

🙏🙏

**Old:** [github-elementor-le-products-caro.zip](https://github.com/elementor/elementor/files/11860194/github-elementor-le-products-caro.zip)

<hr>

**Latest:** [github-elementor-le-products-caro-1.0.1.zip](https://github.com/elementor/elementor/files/11880944/github-elementor-le-products-caro-1.0.1.zip)

Updated it to 1.0.1 ~ Fixed an issue where row and column styling needed to be removed as responsive controls rather than the standard way. Also added checks so the code only runs when Woocommerce, Elementor and Elementor Pro plugins are enabled as it depends on them for this plugin to work so less likely for errors to occur.

**Comment by VerdiH** [comment]
Hi @perks98 we only just started building this small community, not many people know about it YET! To help you out, I;ve just shared your request for help in our main (huge) community on Facebook. See the post here: https://www.facebook.com/groups/Elementors/posts/6067579743367761/

I hope someone will come along soon to help you out. Btw, have you seen this amazing resource yet?
https://developers.elementor.com/

**Reply by perks98** [reply]
Hi @VerdiH, thanks for sharing this on Facebook. I don't really use Facebook much myself so any help is appreciated. 👍

I got a few snippets of info from the developers website but in a few instances it can be lacking. Hopefully the Elementor team will add more notes around Swiper / Carousel instances  😀

**Reply by VerdiH** [reply]
Aside from the Swiper / Carousel instances, please do list anything you feel is missing from our docs. I'm sure @rami-elementor who writes most of them, will be happy with your input 🙏

**Comment by perks98** [comment]
Closing this discussion down. Seems there are no developers or any helpful coding samaritans from the Elementor team that can help with the request. I think perhaps to kick start the developers discussion group would be to have a member from Elementor developers team to help answer questions on the developer vain.
